src/odemis/gui/comp/stream.py

```diff
@@ -813,6 +813,18 @@ def _add_side_label(self, label_text, tooltip=None):
                           flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL, border=5)
         return lbl_ctrl
 
+    @control_bookkeeper
+    def add_immersion_mode_button(self):
+        """ Create and return immersion mode button."""
```

Comment:
I'm a little late at the party, but I wonder about this option: why a toggle button? In Odemis, typically booleans VAs are shown as checkboxes. Why not doing that here too? The rest of the buttons in the GUI are more for actions. This is just a "mode".

Reply:
I didn't know that typically boolean VAs are shown as checkboxes, @tepals do you agree with this suggestion?

Reply:
sounds fine

Reply:
If you go this route, it might be easier to make it part of the stream, as a "emitter_va" (+ an update to the gui.conf.data.STREAM_SETTINGS_CONFIG). You can also do it the current way.